### PR TITLE
[HOTFIX] Credix rebalancing - maths fix for zero-profit case

### PR DESCRIPTION
**Pull request by uxd-vincent**
Turns out having exactly zero profits trips up the computation for the precision loss 😢 

Example: https://explorer.solana.com/tx/5wvq3AK3eViBtgbyhRNQL3TMNxzyVnbr4NJgeeRgtgiCtDb8ybMeG1168GuTzVnEYRKNP83uQ5u8fjxPsxrvVFJg

So this PR adds an additional protection and also adds an integration test to verify that this new version works for exactly zero profits

### CODE REVIEW

File: programs/uxd/tests/integration_tests/suites/test_credix_lp_depository_rebalance_no_profit.rs [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+    // Executing the rebalance request should now work as intended because we are in the execute period
+    program_uxd::instructions::process_rebalance_redeem_withdraw_request_from_credix_lp_depository(
+        &mut program_context,
+        &payer,
+        &collateral_mint.pubkey(),
+        &credix_multisig.pubkey(),
+        &profits_beneficiary_collateral,
+        amount_that_should_remain_liquid - 1, // Available liquidity should be rebalanced (and precision-loss)
+        0,                                    // Zero profit should be available
```
Comment (uxd-vincent) [inline programs/uxd/tests/integration_tests/suites/test_credix_lp_depository_rebalance_no_profit.rs:262]: Tested here

**Review by acamill — APPROVED** [review]

**Review by acamill — APPROVED** [review]
